Kernel/Jail.h

```diff
@@ -37,12 +38,21 @@ class Jail : public RefCounted<Jail> {
     SpinlockProtected<size_t, LockRank::None>& attach_count() { return m_attach_count; }
 
 private:
-    Jail(NonnullOwnPtr<KString>, JailIndex);
+    Jail(NonnullOwnPtr<KString>, JailIndex, NonnullRefPtr<ScopedProcessList>);
 
     NonnullOwnPtr<KString> m_name;
     JailIndex const m_index;
 
-    IntrusiveListNode<Jail, NonnullLockRefPtr<Jail>> m_jail_list_node;
+    IntrusiveListNode<Jail, NonnullLockRefPtr<Jail>> m_list_node;
+
+public:
+    using List = IntrusiveListRelaxedConst<&Jail::m_list_node>;
+
+    NonnullRefPtr<ScopedProcessList> process_list();
+
+private:
```

Comment:
I know we do this other places in the kernel, but having `public:` `private:` `public:` `private:` multiple times in a class makes it harder to identify the public API and the private details. So I would suggest moving these new declarations up into the public section above.